enhancements/audit-command.md

```diff
@@ -0,0 +1,594 @@
+  -w, --wait-time duration       minutes to wait for the report to be complete. Example: 30m (default 20m)                    
+  -o, --output                   output format for results. Valid values: xlsx, json (default "xlsx")
+  -s, --scorecard bool           set false to disable the scorecard checks. (default true). Note that scorecard test requires a cluster running. 
+  -v, --validation bool          set false to disable the bundle validation checks. (default true). 
+  -d, --data                     set false to disable the data report. It is useful for who would like to audit the index via only run the bundle and scorecard checks. (default true).
+  -h, --head-only                set the flag to audit only the operator bundles which are the head of the channel. (default all bundles of the index catalog image will be used)
+  -w, --warning                  set false to disable the report of warning(s) found in the checks (default warning and errors will be reported)
+```
+
+Note that the command will:
+- Use the index image to gathering the data
+- Download the operator bundles
+- Use SDK commands to run the checks 
+- Output the results
+ 
+### User Stories 
+ 
+- As an admin and or maintainer of OLM index catalog, I would like to check if all bundle operators in the catalogue are respecting all expected criteria so that, I can ensure a better quality of all bundles in the catalogue.
+- As an admin and or maintainer of OLM index catalog, I would like to easily know more about the bundles in the catalogue so that, I can easily provide strategies which will be useful for the biggest part of the authors of the projects published on it.
+- As an admin and or maintainer of OLM index catalog, identify problems and concerns in the operators bundles of some catalogue so that, I can easily decide if they should or not be removed from the index catalog.
+- As an admin and or maintainer of OLM index catalog I would like to audit the OLM index catalog by running custom tests for each operator bundle that, I can use it to verify if all operator bundles will work as expected.
+
+### Implementation Details/Notes/Constraints 
+
+Following the actions which are being proposed in this document for the new command:
+
+- Extract the database from the image informed
+- Via sql get all bundles name and paths from the index catalog image
+- Download and extract all bundles files by using the operator bundle path
+- Get the required data for the report via sql from the index, from the operator bundle manifest files and from the project repository
+- Use sdk tool to execute the bundle validate checks
+- Use sdk tool to execute the scorecard bundle checks
+- Output a report providing the information obtained and processed. 
+
+#### Using the index image to gathering the data
+
+Following an example to extract the sql database from the image: 
+
+`oc image extract registry.redhat.io/redhat/redhat-operator-index:v4.6 --file=/database/index.db`
```

Comment:
Use the generic equivalent of this command, since `oc` is for OCP.
````suggestion
```sh
docker create --name rh-catalog registry.redhat.io/redhat/redhat-operator-index:v4.6 "yes"
docker cp rh-catalog:/database/index.db .
```
````

Plus I think `opm` already has some functions to handle this.

Reply:
It is a point to check for sure. Good point I will add in the open questions.